fix: exhaustive-deps warning for DEBEX

TranScripted opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
Fix for the `react-hooks/exhaustive-deps` warning

closes: https://github.com/department-of-veterans-affairs/va.gov-team/issues/110537
## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/110537

## Testing done
Ran `yarn eslint src/applications/disability-benefits/all-claims` to ensure the warning does not appear

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

TranScripted committed:
fix: exhaustive-deps warning for DEBEX

freeheeling reviewed (dismissed):
LGTM. The update correctly lists all variables that the effect depends on.

TranScripted committed:
Merge branch 'main' into debex/fix_exhaustive_dependencies_110539

TranScripted committed:
Merge branch 'main' into debex/fix_exhaustive_dependencies_110539

theodur committed:
Burial benefit logic updates (#37737)

* Add new mappings for BUR benefit

* Add additional burial benefit

* Add missing eligible discharge options

macalemid committed:
Update acceptance criteria for FMP and MHC to allow under honerable discharge (#37825)

macalemid committed:
Ptemsvt 496 improve test quality (#37628)

* Added additional tests to fill in gaps

macalemid committed:
Ptemsvt 499 improve tests 2 (#37789)

* Added additional tests to fill in gaps

flex2016 committed:
VEBT-2614: SCO fix URL link missing letter (#37862)

williamphelps13 committed:
[CST] 107782 Refine loading experience before content renders (#37261)

* Refine loading experience before content renders (no feature-flag version)

* Add tests for loading experience improvements

- Add tests for downtime notification loading states
- Test loading indicator priority (downtime takes precedence over feature flags)
- Update existing tests to use new test IDs (feature-flags-loader)
- Verify va-loading-indicator web component renders correctly
- Test loading message and accessibility attributes

pmclaren19 committed:
112533 chapter 4 tests and field updates (#37846)

* add major as a required field

* Added educationalInstitutionsPages.unit.spec.jsx

* update order of tests

* update tests and add tests for arrayBuilderOptions

* update duplicate tests

jordanBoardmanCodes committed:
MBMS-92618 | Self-Service: PreNeed: Integration: Enhancement - Staging Review finding: va-select imposter component (#37554)

* MBMS-92618 initial commit - a couple of successful updates

* MBMS-92618 updated applicantContactInformation

* MBMS-92618 updated servicePeriodsPages and added a new styling class

* MBMS-92618 updated applicantContactInformation part2

* MBMS-92618 updated sponsorContactInformation

* MBMS-92618 figured out new solution for state field

* MBMS-92618 figured out best solution for dynamic state field function

* MBMS-92618 all AC met - POR/UI/UX ready

* MBMS-92618 fixed/updated failing unit tests

* MBMS-92618 fixed/updated test with redundant code

* MBMS-92618 fixed two failing QA tests

* MBMS-92618 fixed unit tests for applicantMilitaryNameInformation

* MBMS-92618 fixed unit tests for applicantMilitaryNameInformationPreparer

flex2016 committed:
VEBT-2478: Form 22-10297 Step 1 Page 6: Date released from active duty (#37859)

* VEBT-2478: Form 22-10297 Step 1 Page 6: Date released from active duty

* add missing period

TranScripted committed:
fix unit test

longmd committed:
10-10EZ | Populate Dependents ArrayBuilder Config (#37871)
